internal/io/io.go

```diff
@@ -72,6 +72,23 @@ func JSONRoundTrip(from any, to any) error {
 	return json.Unmarshal(bs, to) //nolint:wrapcheck
 }
 
+// YAMLRoundTrip convert any value to YAML and back again.
+func YAMLRoundTrip(from any, to any) error {
```

Comment:
While I am not very familiar with (un)marshalling YAML, I do not immediately see how making a roundtrip via YAML as the intermediate is any different from JSON when considering the end result. Or is there some go types that would behave differently in this case?

Reply:
No, you're right. This remained from a point where I considered ditching JSON altogether, but I realized later we need that for output formats. I'll have this removed before merge.